### PR TITLE
[arte] Allow downloading age restricted content

### PR DESCRIPTION
**Pull request by StefanLobbenmeier**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

 
I thought that arte.tv looks at the cookies to determine if the logged in user has confirmed their age. But to my surprise, the cookies are not sent in the request, instead there is a header called x-validated-age that is set by the client. When I set that client in yt-dlp we can also access age restricted content.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [ ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
🎥🔞🌐

Add `X-Arte-User-Profile` header to `arte.py` extractor to access age-restricted videos. This fixes the extraction of some Arte videos that are blocked by age verification.

> _`config` header set_
> _Bypassing age check for Arte_
> _Winter of content_

### Walkthrough
* Bypass age verification for some Arte videos by adding a header to the config JSON request ([link](https://github.com/yt-dlp/yt-dlp/pull/8301/files?diff=unified&w=0#diff-fb227514252628a6a5c79bbbaff82b742fe714fd910e068d9a23d854838743adL139-R143))

</details>

### CODE REVIEW

**Comment by StefanLobbenmeier** [comment]
Related issue: https://github.com/yt-dlp/yt-dlp/issues/7782

**Comment by StefanLobbenmeier** [comment]
Request in browser:
image

File: yt_dlp/extractor/arte.py [hunk 1]
```diff
@@ -136,7 +126,9 @@ def _real_extract(self, url):
         lang = mobj.group('lang') or mobj.group('lang_2')
         langauge_code = self._LANG_MAP.get(lang)
 
-        config = self._download_json(f'{self._API_BASE}/config/{lang}/{video_id}', video_id)
+        config = self._download_json(f'{self._API_BASE}/config/{lang}/{video_id}', video_id, headers={
+            "x-validated-age": 18
```
Comment (bashonly) [inline yt_dlp/extractor/arte.py:130]: Single quotes
And core code fixes this, but all header values should be strings
```suggestion
            'x-validated-age': '18'
```

**Review by bashonly — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/arte.py [hunk 2]
```diff
@@ -67,19 +57,19 @@ class ArteTVIE(ArteTVBaseIE):
         'only_matching': True,
     }, {
         'url': 'https://www.arte.tv/de/videos/110203-006-A/zaz/',
+        'only_matching': True,
+    }, {
+        'url': 'https://www.arte.tv/de/videos/006785-000-A/the-element-of-crime/',
```
Comment (bashonly) [inline yt_dlp/extractor/arte.py:62]:
```suggestion
    }, {
        'note': 'age-restricted',
        'url': 'https://www.arte.tv/de/videos/006785-000-A/the-element-of-crime/',
```

**Review by bashonly — APPROVED** [review]

**Pull request merged**